diff -Naurp net-tools-1.60/ether-wake.c net-tools-1.60.oden/ether-wake.c
--- net-tools-1.60/ether-wake.c	2008-12-22 12:20:51.000000000 +0100
+++ net-tools-1.60.oden/ether-wake.c	2008-12-22 12:21:08.000000000 +0100
@@ -131,7 +131,7 @@ int main(int argc, char *argv[])
 		case 'D': debug++;			break;
 		case 'i': ifname = optarg;	break;
 		case 'p': get_wol_pw(optarg); break;
-		case 'u': printf(usage_msg); return 0;
+		case 'u': printf("%s", usage_msg); return 0;
 		case 'v': verbose++;		break;
 		case 'V': do_version++;		break;
 		case '?':
@@ -140,7 +140,7 @@ int main(int argc, char *argv[])
 	if (verbose || do_version)
 		printf("%s\n", version_msg);
 	if (errflag) {
-		fprintf(stderr, brief_usage_msg);
+		fprintf(stderr, "%s", brief_usage_msg);
 		return 3;
 	}
 
diff -Naurp net-tools-1.60/mii-diag.c net-tools-1.60.oden/mii-diag.c
--- net-tools-1.60/mii-diag.c	2008-12-22 12:20:51.000000000 +0100
+++ net-tools-1.60.oden/mii-diag.c	2008-12-22 12:21:08.000000000 +0100
@@ -210,7 +210,7 @@ main(int argc, char **argv)
 	}
 
 	if (verbose || opt_version)
-		printf(version);
+		printf("%s", version);
 
 	/* Open a basic socket. */
 	if ((skfd = socket(AF_INET, SOCK_DGRAM,0)) < 0) {
@@ -538,7 +538,7 @@ int show_basic_mii(long ioaddr, int phy_
 			   bmcr & 0x0100 ? "full":"half");
 	for (i = 0; i < 9; i++)
 		if (bmcr & (0x0080<<i))
-			printf(bmcr_bits[i]);
+			printf("%s", bmcr_bits[i]);
 
 	new_bmsr = mdio_read(ioaddr, phy_id, 1);
 	if ((bmsr & 0x0016) == 0x0004)
