diff -Naurp net-tools-1.60/ether-wake.c net-tools-1.60.oden/ether-wake.c
--- net-tools-1.60/ether-wake.c	2008-12-22 12:20:51.000000000 +0100
+++ net-tools-1.60.oden/ether-wake.c	2008-12-22 12:21:08.000000000 +0100
@@ -131,7 +131,7 @@ int main(int argc, char *argv[])
 		case 'D': debug++;			break;
 		case 'i': ifname = optarg;	break;
 		case 'p': get_wol_pw(optarg); break;
-		case 'u': printf(usage_msg); return 0;
+		case 'u': printf("%s", usage_msg); return 0;
 		case 'v': verbose++;		break;
 		case 'V': do_version++;		break;
 		case '?':
@@ -140,7 +140,7 @@ int main(int argc, char *argv[])
 	if (verbose || do_version)
 		printf("%s\n", version_msg);
 	if (errflag) {
-		fprintf(stderr, brief_usage_msg);
+		fprintf(stderr, "%s", brief_usage_msg);
 		return 3;
 	}
 
diff -Naurp net-tools-1.60/mii-diag.c net-tools-1.60.oden/mii-diag.c
--- net-tools-1.60/mii-diag.c	2008-12-22 12:20:51.000000000 +0100
+++ net-tools-1.60.oden/mii-diag.c	2008-12-22 12:21:08.000000000 +0100
@@ -210,7 +210,7 @@ main(int argc, char **argv)
 	}
 
 	if (verbose || opt_version)
-		printf(version);
+		printf("%s", version);
 
 	/* Open a basic socket. */
 	if ((skfd = socket(AF_INET, SOCK_DGRAM,0)) < 0) {
@@ -538,7 +538,7 @@ int show_basic_mii(long ioaddr, int phy_
 			   bmcr & 0x0100 ? "full":"half");
 	for (i = 0; i < 9; i++)
 		if (bmcr & (0x0080<<i))
-			printf(bmcr_bits[i]);
+			printf("%s", bmcr_bits[i]);
 
 	new_bmsr = mdio_read(ioaddr, phy_id, 1);
 	if ((bmsr & 0x0016) == 0x0004)
diff -Naurp net-tools-1.60/mii-tool.c net-tools-1.60.oden/mii-tool.c
--- net-tools-1.60/mii-tool.c	2008-12-22 12:20:50.000000000 +0100
+++ net-tools-1.60.oden/mii-tool.c	2008-12-22 12:21:08.000000000 +0100
@@ -241,7 +241,7 @@ int show_basic_mii(int sock, int phy_id)
 
     if (opt_watch) {
 	if (opt_log) {
-	    syslog(LOG_INFO, buf);
+	    syslog(LOG_INFO, "%s", buf);
 	} else {
 	    char s[20];
 	    time_t t = time(NULL);
@@ -419,7 +419,7 @@ int main(int argc, char **argv)
     }
 
     if (opt_version)
-	printf(version);
+	printf("%s", version);
 
     /* Open a basic socket. */
     if ((skfd = socket(AF_INET, SOCK_DGRAM,0)) < 0) {
diff -Naurp net-tools-1.60/rarp.c net-tools-1.60.oden/rarp.c
--- net-tools-1.60/rarp.c	2001-04-08 19:05:05.000000000 +0200
+++ net-tools-1.60.oden/rarp.c	2008-12-22 12:21:08.000000000 +0100
@@ -225,7 +225,7 @@ int main(int argc, char **argv)
 	case 'h':
 	    usage();
 	case 'V':
-	    fprintf(stderr, version_string);
+	    fprintf(stderr, "%s", version_string);
 	    exit(E_VERSION);
 	    break;
 	case 'v':
diff -Naurp net-tools-1.60/slattach.c net-tools-1.60.oden/slattach.c
--- net-tools-1.60/slattach.c	2000-10-28 12:59:41.000000000 +0200
+++ net-tools-1.60.oden/slattach.c	2008-12-22 12:21:08.000000000 +0100
@@ -565,7 +565,7 @@ usage(void)
 	  "[-c cmd] [-s speed] [-p protocol] tty | -\n"
 	  "       slattach -V | --version\n";
 
-  fprintf(stderr, usage_msg);
+  fprintf(stderr, "%s", usage_msg);
   exit(1);
 }
 
